src/averageAges.js

```diff
@@ -55,11 +82,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
-}
+  const objWithMom = people
+    .filter(({ sex }) => sex === 'f');
+
+  const objWithChildren = people
+    .filter(({ mother }) => mother !== null);
+
+  const objWithSon = people
+    .filter(({ mother, sex }) => mother !== null && sex === 'm');
```

Comment:
You have already filtered the array of people and saved a new array that definitely contains mothers, so you don't need to do it again.